app/lib/csv_importer.rb

```diff
@@ -1,41 +1,56 @@
-module CsvImporter
-  CSV_CATEGORIES = {
-      SPAWNING_SUCCESS: 'Spawning Success',
-      TAGGED_ANIMAL_ASSESSMENT: 'Tagged Animal Assessment',
-      UNTAGGED_ANIMAL_ASSESSMENT: 'Untagged Animal Assessment'
-  }.freeze
+class CsvImporter
+  attr_reader :stats, :model, :filename, :processed_file_id
 
-  class InvalidCsvCategoryError < StandardError; end;
+  CATEGORIES = [
+      'Spawning Success',
+      'Tagged Animal Assessment',
+      'Untagged Animal Assessment',
+      'Wild Collection',
+  ].freeze
+
+  class InvalidCategoryError < StandardError; end;
 
   def self.import(filename, category_name, processed_file_id)
-    csv_category_model = model_from_category(category_name)
-    IOStreams.each_record(filename) do |record|
-      attrs = translate_attribute_names(record, category_name)
-      attrs[:processed_file_id] = processed_file_id
-      csv_row_record = csv_category_model.new(attrs.merge({raw: false}))
-      csv_row_record.cleanse_data! if csv_row_record.respond_to?(:cleanse_data!)
-      csv_row_record.save
+    importer = new(filename, category_name, processed_file_id)
```

Comment:
I like instances of things, so I converted the previous module method to a class method that creates an instance of an importer to perform the work. The "API" of using the `CsvImporter.import()` method still feels the same in the places it is used.

Reply:
I like this too. Using instances feels much more OO to me.